diagnostics/diag.go

```diff
@@ -98,7 +98,18 @@ func (di *DiagInfo) Marshal() []byte {
 }
 
 func (d *Diagnostics) getPeers() []peer.ID {
-	return d.host.Network().Peers()
+	peers := d.host.Network().Peers()
+	pmap := make(map[peer.ID]struct{})
```

Comment:
s/pmap/seen (it's helpful for a variable to carry meaning/help to explain the task at hand)